Gentoo bug 66557
Link: http://bugs.gentoo.org/show_bug.cgi?id=66557
Patch-By: REdOG

Submitted upstream: No

The uclibc environment, as opposed to the glibc environment, needs explicit 
linking with the pthread & dl libraries. Without these, name resolution is 
not functional.

--- asterisk-1.8.7.0-rc2.orig/Makefile	2011-08-25 21:00:55.000000000 +0200
+++ asterisk-1.8.7.0-rc2.orig/Makefile	2011-09-21 14:13:12.823076046 +0200
@@ -247,6 +247,10 @@ 
   _ASTCFLAGS+=-pthread -ftrampolines
 endif
 
+ifeq ($(OSARCH),linux-uclibc)
+  AST_LIBS+=-lpthread -ldl
+endif
+
 ifeq ($(OSARCH),SunOS)
   _ASTCFLAGS+=-Wcast-align -DSOLARIS -I../include/solaris-compat -I/opt/ssl/include -I/usr/local/ssl/include -D_XPG4_2 -D__EXTENSIONS__
 endif
